plugins/inputs/kube_lite/kube_lite.go

```diff
@@ -0,0 +1,213 @@
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// KubernetesState represents the config object for the plugin.
+type KubernetesState struct {
+	URL string
+
+	// Bearer Token authorization file path
+	BearerToken string `toml:"bearer_token"`
+
+	// MaxConnections for worker pool tcp connections
+	MaxConnections int `toml:"max_connections"`
+
+	// HTTP Timeout specified as a string - 3s, 1m, 1h
+	ResponseTimeout internal.Duration `toml:"response_timeout"`
+
+	tls.ClientConfig
+
+	client          *client
+	rListHash       string
+	filter          filter.Filter
+	lastFilterBuilt time.Time
+	// try to collect everything on first run
+	firstTimeGather           bool
+	ResourceListCheckInterval *internal.Duration `toml:"resouce_list_check_interval"`
+	ResourceExclude           []string           `toml:"resource_exclude"`
+}
+
+var sampleConfig = `
+  ## URL for the kubelet
+  url = "http://1.1.1.1:10255"
+
+  ## Use bearer token for authorization
+  #  bearer_token = /path/to/bearer/token
+
+  ## Set response_timeout (default 5 seconds)
+  #  response_timeout = "5s"
+
+  ## Worker pool for kube_state_metric plugin only
+  #  empty this field will use default value 30
+  #  max_connections = 5
+
+  ## Optional Resources to exclude from gathering
+  ## Leave them with blank with try to gather everything available.
+  ## Values can be "deployments", "nodes", "pods", "statefulsets"
+  #  resource_exclude = [ "deployments", "nodes", "statefulsets" ]
+
+  ## Optional Resouce List Check Interval, leave blank will use the default
+  #  value of 1 hour. This is the interval to check available resource lists.
+  #  resouce_list_check_interval = "1h"
+
+  ## Optional TLS Config
+  #  tls_ca = /path/to/cafile
+  #  tls_cert = /path/to/certfile
+  #  tls_key = /path/to/keyfile
+  ## Use TLS but skip chain & host verification
+  #  insecure_skip_verify = false
+`
+
+// SampleConfig returns a sample config
+func (ks *KubernetesState) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the description of this plugin
+func (ks *KubernetesState) Description() string {
+	return "Read metrics from the kubernetes kubelet api"
+}
+
+// Gather collects kubernetes metrics from a given URL.
+// todo: convert to service?
+func (ks *KubernetesState) Gather(acc telegraf.Accumulator) (err error) {
+	var rList *metav1.APIResourceList
+	var cutoff time.Time
+	if ks.client == nil {
+		if ks.client, rList, err = ks.initClient(); err != nil {
+			return err
+		}
+		goto buildFilter
+	}
+
+	cutoff = time.Now().Add(-1 * ks.ResourceListCheckInterval.Duration)
+
+	// Here we just test
+	if ks.lastFilterBuilt.Unix() > 0 && ks.lastFilterBuilt.Before(cutoff) {
+		goto doGather
+	}
+
+	rList, err = ks.client.getAPIResourceList(context.Background())
```

Comment:
Make one context at top of Gather and use it as the root of all other contexts.  In the future Telegraf should provide a context to the Gather function.